packages/victory-core/src/victory-util/data.js

```diff
@@ -91,6 +92,72 @@ function cleanData(dataset, props) {
   });
 }
 
+// This method will remove data points that fall outside of the desired domain (non-continuous charts only)
+function formatDataFromDomain(dataset, props) {
+  const { domain, symbol, interpolation } = props;
+
+  if (!domain || interpolation) return dataset;
+
+  const domainX = domain.x || Domain.getDomainFromData(props, "x", dataset);
```

Comment:
Thinking about this more, we shouldn't need to calculate `Domain.getDomainFromData`, because there will never be values _outside_ of that. If there isn't a domain given by props for a given axis, we won't ever be nulling values. That's a relatively expensive check, so we should definitely avoid it.

Reply:
So I actually only added `Domain.getDomainFromData` because the karma tests in the domain spec fail without it.  Because the tests don't run `getCalculatedProps` beforehand, they're just supplying some shallow dummy data that isn't complete enough for my new function to run completely, domain hasn't been fully calculated.  I have yet to see `Domain.getDomainFromData` called in the normal flow of components outside of those tests.  domain.x and domain are always set and those get used instead of calling out to the expensive function.  

Is there another way I can solve this that you know of offhand that would satisfy those tests but not have to resort to using `Domain.getDomainFromData`?

Reply:
I could potentially return dataset early if domain.x and domain.y aren't set, like where I check for domain itself to be set.

Reply:
After a lot of testing I went ahead and did this and removed the Domain.getDomainFromData logic here.